### ISSUE TITLE
[Request] Nearest-Neighbor Scaling

### ISSUE THREAD

**Issue by saphtea** [bubble2k16/snes9x_3ds#38]
Hey! Just something I noticed is when you choose the "Pixel Perfect fit" to screen option that it seems to be scaling it with a linear algorithm which makes things a bit blurry. Would you be able to allow an option for scaling by Nearest Neighbor instead by any chance?

**Comment by bubble2k16** [comment]
Any game in particular has this problem? Pixel perfect is 1:1 and shouldn't exhibit any scaling issue.

**Comment by FateForWindows** [comment]
Pretty sure he means Pixel Perfect Fit, not Pixel Perfect.

**Comment by saphtea** [comment]
Yeah pixel perfect fit sorry about that! Had a misplaced end quote haha. Pixel perfect is 1:1 and doesn't have any issues, but the pixel perfect fit does introduce bluriness (most likely to using a bilenear scaling as opposed to the nearest neighbor scaling). Sorry about that!

**Comment by d3xMachina** [comment]
I'm having the same issue. When using pixel perfect fit it become very blurry so I can't really use it. It would be a great addition to have a better scaling algorithm which is sharper. It's the only thing that is missing for me to make this emulator perfect.

**Comment by bubble2k16** [comment]
I actually tried nearest neighbour, but it seems like there are some inaccuracies in the 3D hardware with floating point math, the screen ends up having lots jagged edges at the bottom-right quadrant of the top screen. I don't quite know how to solve that, so I aborted that feature for now.

**Comment by LoganLieou** [comment]
is this inactive?
